kernel/panic.c

```diff
@@ -343,7 +343,7 @@ void panic(const char *fmt, ...)
 #endif
 
 #ifdef CONFIG_CARTESI_HALT_ON_PANIC
-machine_halt();
+	machine_restart("1");
```

Comment:
I have a question, what will happen when running a kernel with QEMU instead of our emulator?

Reply:
I had a quick look at this and both `poweroff` (without `-f`) and `xhalt` cause qemu to loop reboots.
`poweroff -f` works as expected.

I'll have a look at the OpenSBI qemu is using.
And what is the actual difference between `poweroff` with and without `-f`.